src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/OrdinalStringFrozenDictionary.cs

```diff
@@ -28,52 +32,46 @@ internal sealed class OrdinalStringFrozenDictionary<TValue> : FrozenDictionary<s
             var entries = new KeyValuePair<string, TValue>[source.Count];
             ((ICollection<KeyValuePair<string, TValue>>)source).CopyTo(entries, 0);
 
-            _keys = new string[entries.Length];
+            _keys = keys;
             _values = new TValue[entries.Length];
+            _minimumLength = minimumLength;
+            _maximumLengthDiff = maximumLengthDiff;
 
-            _partialComparer = ComparerPicker.Pick(
-                Array.ConvertAll(entries, pair => pair.Key),
-                ignoreCase: ReferenceEquals(comparer, StringComparer.OrdinalIgnoreCase),
-                out _minimumLength,
-                out _maximumLengthDiff);
+            HashIndex = hashIndex;
+            HashCount = hashCount;
 
             _hashTable = FrozenHashTable.Create(
                 entries,
-                pair => _partialComparer.GetHashCode(pair.Key),
+                pair => GetHashCode(pair.Key),
                 (index, pair) =>
                 {
                     _keys[index] = pair.Key;
                     _values[index] = pair.Value;
                 });
         }
 
-        /// <inheritdoc />
+        private protected int HashIndex { get; }
+        private protected int HashCount { get; }
+        private protected abstract bool Equals(string? x, string? y);
+        private protected abstract int GetHashCode(string s);
         private protected override string[] KeysCore => _keys;
-
-        /// <inheritdoc />
         private protected override TValue[] ValuesCore => _values;
-
-        /// <inheritdoc />
         private protected override Enumerator GetEnumeratorCore() => new Enumerator(_keys, _values);
-
-        /// <inheritdoc />
         private protected override int CountCore => _hashTable.Count;
 
-        /// <inheritdoc />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
It'd be great to add a comment about the pattern being used here.  It'd be really easy for someone to come along and see the overrides that just delegate to the base and think that's silly and remove it.